common/src/test/java/org/apache/uniffle/common/netty/protocol/NettyProtocolTestUtils.java

```diff
@@ -0,0 +1,119 @@
+      super(shuffleId, partitionId, blockId, length, crc, data, shuffleServerInfos, uncompressLength, freeMemory, taskAttemptId);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      ShuffleBlockInfoTest that = (ShuffleBlockInfoTest) o;
+      return this.getPartitionId() == that.getPartitionId()
+                 && this.getBlockId() == that.getBlockId()
+                 && this.getLength() == that.getLength()
+                 && this.getShuffleId() == that.getShuffleId()
+                 && this.getCrc() == that.getCrc()
+                 && this.getTaskAttemptId() == that.getTaskAttemptId()
+                 && this.getUncompressLength() == that.getUncompressLength()
+                 && this.getFreeMemory() == that.getFreeMemory()
+                 && this.getData().equals(that.getData())
+                 && this.getShuffleServerInfos().equals(that.getShuffleServerInfos());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(this.getPartitionId(), this.getBlockId(), this.getLength(), this.getShuffleId(),
+          this.getCrc(), this.getTaskAttemptId(), this.getData(), this.getShuffleServerInfos(),
+          this.getUncompressLength(), this.getFreeMemory());
+    }
+  }
+
+  private static SendShuffleDataRequestTest toSendShuffleDataRequestTest(SendShuffleDataRequest sendShuffleDataRequest) {
+    return new SendShuffleDataRequestTest(sendShuffleDataRequest.requestId,
+        sendShuffleDataRequest.getAppId(),
+        sendShuffleDataRequest.getShuffleId(),
+        sendShuffleDataRequest.getRequireId(),
+        sendShuffleDataRequest.getPartitionToBlocks(),
+        sendShuffleDataRequest.getTimestamp());
+  }
+
+  private static List<ShuffleBlockInfoTest> toShuffleBlockInfoTestList(List<ShuffleBlockInfo> shuffleBlockInfoList) {
+    List<ShuffleBlockInfoTest> res = Lists.newArrayList();
+    for (ShuffleBlockInfo shuffleBlockInfo : shuffleBlockInfoList) {
+      res.add(toShuffleBlockInfoTest(shuffleBlockInfo));
+    }
+    return res;
+  }
+
+  private static ShuffleBlockInfoTest toShuffleBlockInfoTest(ShuffleBlockInfo shuffleBlockInfo) {
+    return new ShuffleBlockInfoTest(shuffleBlockInfo.getShuffleId(),
+        shuffleBlockInfo.getPartitionId(),
+        shuffleBlockInfo.getBlockId(),
+        shuffleBlockInfo.getLength(),
+        shuffleBlockInfo.getCrc(),
+        shuffleBlockInfo.getData(),
+        shuffleBlockInfo.getShuffleServerInfos(),
+        shuffleBlockInfo.getUncompressLength(),
+        shuffleBlockInfo.getFreeMemory(),
+        shuffleBlockInfo.getTaskAttemptId());
+  }
+
+  public static boolean compareSendShuffleDataRequest(SendShuffleDataRequest req1, SendShuffleDataRequest req2) {
```

Comment:
Class SendShuffleDataRequestTest seems unnecessary.
```
return req1.getPartitionId() == req2.getPartitionId()
                 &&  req1.getBlockId() == req2.getBlockId()
                 && req1.getLength() == req2.getLength()
                 && req1.getShuffleId() == req2.getShuffleId()
                 && req1.getCrc() == req2.getCrc()
                 && req1.getTaskAttemptId() == req2.getTaskAttemptId()
                 && req1.getUncompressLength() == req2.getUncompressLength()
                 && req1.getFreeMemory() == req2.getFreeMemory()
                 && req1.getData().equals(req2.getData())
                 && CompareShuffleInfos(req1.getShuffleServerInfos(), req2.getShuffleServerInfos());
```

Reply:
> Class SendShuffleDataRequestTest seems unnecessary.
> 
> ```
> return req1.getPartitionId() == req2.getPartitionId()
>                  &&  req1.getBlockId() == req2.getBlockId()
>                  && req1.getLength() == req2.getLength()
>                  && req1.getShuffleId() == req2.getShuffleId()
>                  && req1.getCrc() == req2.getCrc()
>                  && req1.getTaskAttemptId() == req2.getTaskAttemptId()
>                  && req1.getUncompressLength() == req2.getUncompressLength()
>                  && req1.getFreeMemory() == req2.getFreeMemory()
>                  && req1.getData().equals(req2.getData())
>                  && CompareShuffleInfos(req1.getShuffleServerInfos(), req2.getShuffleServerInfos());
> ```

If we don't rewrite the equals method, we need to write a bunch of redundant methods, such as list, compare of map content.

Reply:
It's ok to add the compare method of list and map.